File: selectionSort.cpp
#include <iostream>
#include <fstream>
#include <ctime>
#include <iomanip>

using namespace std;

void selectionSort(int arr[], int n) {
    for (int i = 0; i < n - 1; ++i) {
        int minIdx = i;
        for (int j = i + 1; j < n; ++j) {
            if (arr[j] < arr[minIdx]) {
                minIdx = j;
            }
        }
        swap(arr[i], arr[minIdx]);
    }
}

int importNumbersFromFile(const string& filePath, int arr[], int maxSize) {
    ifstream file(filePath);
    int num, i = 0;
    
    while (file >> num && i < maxSize) {
        arr[i++] = num;
    }

    return i;
}

void measureTime(const string& filePath) {
    const int maxSize = 1000000;
    int numbers[maxSize];
    int size = importNumbersFromFile(filePath, numbers, maxSize);

    clock_t startTime = clock();

    selectionSort(numbers, size);

    clock_t endTime = clock();

    double diff = double(endTime - startTime) / CLOCKS_PER_SEC;
    
    cout << fixed << setprecision(5);
    cout << "processing time: " << diff << " second" << endl;

    cout << "Sorted list (first 5): ";
    for (int i = 0; i < 10 && i < size; ++i) {
        cout << numbers[i] << " ";
    }
    cout << endl;
}

int main() {
    string filePath = "dataset.txt";
    measureTime(filePath);
    return 0;
}

File: selectionSort.py
import time

def selection_sort(arr):
    n = len(arr)
    for i in range(n):
        for j in range(i + 1, n):
            if arr[j] < arr[i]:
                arr[i], arr[j] = arr[j], arr[i]

def import_numbers_from_file(file_path):
    with open(file_path, 'r') as file:
        numbers = list(map(int, file.read().split()))
    return numbers

def measure_time(file_path):
    numbers = import_numbers_from_file(file_path)
    
    start_time = time.time()
    selection_sort(numbers)
    end_time = time.time()

    print(f"Sorted list: {numbers[:5]}")
    print(f"processing time: {end_time - start_time:.3f} second")


measure_time("dataset.txt")
